PayoutCurve.md

```diff
@@ -67,54 +69,113 @@ In this section we detail the TLV serialization for a general `payout_function`.
 
 1. type: 42790 (`payout_function_v0`)
 2. data:
-   * [`u16`:`num_pts`]
-   * [`boolean`:`is_endpoint_1`]
-   * [`bigsize`:`event_outcome_1`]
-   * [`bigsize`:`outcome_payout_1`]
-   * [`u16`:`extra_precision_1`]
+   * [`u16`:`num_pieces`]
+   * [`bigsize`:`endpoint_0`]
+   * [`bigsize`:`endpoint_payout_0`]
+   * [`u16`:`extra_precision_0`]
+   * [`payout_curve_piece`:`piece_1`]
+   * [`bigsize`:`endpoint_1`]
    * ...
-   * [`boolean`:`is_endpoint_num_pts`]
-   * [`bigsize`:`event_outcome_num_pts`]
-   * [`bigsize`:`outcome_payout_num_pts`]
-   * [`u16`:`extra_precision_num_pts`]
-
-`num_pts` is the number of points on the payout curve that will be provided for interpolation.
-Each point consists of a `boolean` and two `bigsize` integers and a `u16`.
-
-The `boolean` is called `is_endpoint` and if this is true, then this point marks the end of a
-polynomial piece and the beginning of a new one. If this is false then this is a midpoint
-between the previous and next endpoints.
-The first integer is called `event_outcome` and contains the actual number that could be signed by the oracle
-(note: not in the serialization used by the oracle) which corresponds to an x-coordinate on the payout curve.
-The second integer is called `outcome_payout` and is set equal to the local party's payout should
-`event_outcome` be signed which corresponds to a y-coordinate on the payout curve.
+   * [`payout_curve_piece`:`piece_num_pieces`]
+   * [`bigsize`:`endpoint_num_pieces`]
+   * [`bigsize`:`endpoint_payout_num_pieces`]
+   * [`u16`:`extra_precision_num_pieces`]
+
+`num_pieces` is the number of `payout_curve_pieces` which make up the payout curve along with their endpoints.
+Each endpoint consists of a two `bigsize` integers and a `u16`.
+
+The first integer is called `endpoint` and contains the actual `event_outcome` which corresponds to an x-coordinate
+on the payout curve which is a boundary between curve pieces.
+The second integer is called `endpoint_payout` and is set equal to the local party's payout should
+`endpoint` be signed which corresponds to a y-coordinate on the payout curve.
```

Comment:
Nit: phrasing is a bit confusing to me. Maybe: "local party's payout for the corresponding `enpoint` outcome"?